File: CompilerRR/CompilerRR/src/ch/fhnw/cpib/compiler/classes/OptionalProgramParameters.java
package ch.fhnw.cpib.compiler.classes;

import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalCHANGEMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalFLOWMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IRepeatingOptionalProgramParameters;
import ch.fhnw.cpib.compiler.scanner.Token;

public class OptionalProgramParameters implements IConcSyn.IOptionalProgramParameters {

	IConcSyn.IOptionalFLOWMODE flowMode;
	IConcSyn.IOptionalCHANGEMODE changeMode;
	IConcSyn.ITypedIdent typedIdent;
	IConcSyn.IRepeatingOptionalProgramParameters repOptProPar;
	public OptionalProgramParameters(IOptionalFLOWMODE flowMode, IOptionalCHANGEMODE changeMode, IConcSyn.ITypedIdent typedIdent,
			IRepeatingOptionalProgramParameters repOptProPar) {
		super();
		this.flowMode = flowMode;
		this.changeMode = changeMode;
		this.typedIdent = typedIdent;
		this.repOptProPar = repOptProPar;
	}
	
	
}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/OptionalProgramParameters.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class OptionalProgramParametersEps : IOptionalProgramParameters
    {
        public IAbsList<AbsProgramParameter> ToAbySyn()
        {
            return new AbsEmptyList<AbsProgramParameter>();
        }
    }

    class OptionalProgramParametersParams : IOptionalProgramParameters
    {
        public IOptionalFlowMode FlowMode { get; private set; }
        public IOptionalChangeMode ChangeMode { get; private set; }
        public ITypedIdent TypedIdent { get; private set; }
        public IRepeatingOptionalProgramParameters OptionalProgramParameters { get; private set; }

        public OptionalProgramParametersParams (IOptionalFlowMode flowMode, IOptionalChangeMode changeMode,
            ITypedIdent typedIdent, IRepeatingOptionalProgramParameters progParams)
        {
            FlowMode = flowMode;
            ChangeMode = changeMode;
            TypedIdent = typedIdent;
            OptionalProgramParameters = progParams;
        }

        public IAbsList<AbsProgramParameter> ToAbySyn()
        {
            var identInfo = TypedIdent.ToAbySyn();
            return new AbsProgramParameterList()
            {
                ProgramParameter =
                    new AbsProgramParameter()
                    {
                        ChangeMode = this.ChangeMode.ToAbySyn(),
                        FlowMode = this.FlowMode.ToAbsSyn(),
                        Ident = identInfo.Ident,
                        Type = identInfo.Type
                    },
                ProgramParameterList = OptionalProgramParameters.ToAbySyn()
            };
        }
    }
}
